### PR TITLE
Identifiers are now real types

### PR DESCRIPTION
**Pull request by cmyr**
This patch makes better use of the type system to distinguish between `BufferIdentifier` and `ViewIdentifier`. 

There were a few spots in the plugin code where this was poorly distinguished, and this change makes me a bit more considered about those places. It also will make it easier to use some value type as an identifier in the future.

### CODE REVIEW

File: rust/core-lib/src/plugins/manager.rs [hunk 1]
```diff
@@ -110,43 +112,67 @@ impl <W: Write + Send + 'static>PluginManager<W> {
...
+                    Ok(plugin_ref) => {
+                        plugin_ref.initialize(&init_info);
+                        let mut inner = me.lock();
+                        let mut running = false;
+
+                        if let Some(ed) = inner.buffers.lock()
+                            .editor_for_view(&view_id2) {
+                                ed.plugin_started(&view_id2, &key.1);
+                                running = true;
+                            }
```
Comment (trishume) [inline rust/core-lib/src/plugins/manager.rs]: This close paren is mis-indented.

But more importantly, why did you change this from how it was before to add the `running` variable? Since `running` only becomes true in the `if` block, why not remove the `running` variable and put `inner.running.insert(key, plugin_ref)` inside the `if` block.

Comment (cmyr) [inline rust/core-lib/src/plugins/manager.rs]: hmm, I was doing it this way because `inner.running.insert` is a mutable borrow, and we have an immutable borrow out when we take `editor_for_view`. It's possible I'm missing something. Just moving it into the `if` block doesn't compile.

My editor wants to align that closing brace with the second line of the opening expression. I don't really have an opinion here, will change, but it might get reverted next time I fix whitespace 😐.

Comment (trishume) [inline rust/core-lib/src/plugins/manager.rs]: Darn borrow checker. Having to do dumb things like that for no good reason when using mutable borrows is one of my least favourite things about Rust.
